### PR TITLE
Fix 4263 multiband raster

### PR DESCRIPTION
**Pull request by ismailsunni**
### What does it fix?
* Ticket: #4263 #3505 
* Funded by: DFAT
* Description: 
   - Adding new keyword property for active band
   - Update wizard to be able to assign active band
   - Unit test

### CODE REVIEW

File: safe/gui/tools/wizard/step_kw13_band_selector.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+"""InaSAFE Keyword Wizard Band Selector."""
+
+# noinspection PyPackageRequirements
+from PyQt4 import QtCore
+from PyQt4.QtGui import QListWidgetItem
+
+from qgis.core import QgsRasterBandStats
+
+from safe.utilities.i18n import tr
+from safe.common.exceptions import InvalidWizardStep
```
Comment (ismailsunni) [inline safe/gui/tools/wizard/step_kw13_band_selector.py]: __[flake8]__
* [F401] 'safe.common.exceptions.InvalidWizardStep' imported but unused

File: safe/gui/tools/wizard/step_kw13_band_selector.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+from safe.gui.tools.wizard.wizard_step import (
+    get_wizard_step_ui_class, WizardStep)
+from safe.gui.tools.wizard.wizard_strings import (
```
Comment (ismailsunni) [inline safe/gui/tools/wizard/step_kw13_band_selector.py]: __[flake8]__
* [F401] 'safe.gui.tools.wizard.wizard_strings.layer_mode_raster_question' imported but unused
* [F401] 'safe.gui.tools.wizard.wizard_strings.layer_mode_vector_classified_confirm' imported but unused
* [F401] 'safe.gui.tools.wizard.wizard_strings.layer_mode_vector_question' imported but unused
* [F401] 'safe.gui.tools.wizard.wizard_strings.layer_mode_vector_continuous_confirm' imported but unused

File: safe/gui/tools/wizard/step_kw13_band_selector.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    layer_mode_raster_question,
+    layer_mode_vector_question,
+    layer_mode_vector_classified_confirm,
+    layer_mode_vector_continuous_confirm)
+from safe.utilities.gis import is_raster_layer
```
Comment (ismailsunni) [inline safe/gui/tools/wizard/step_kw13_band_selector.py]: __[flake8]__
* [F401] 'safe.utilities.gis.is_raster_layer' imported but unused

File: safe/gui/tools/wizard/step_kw13_band_selector.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+from safe.definitions.utilities import definition, get_layer_modes
```
Comment (ismailsunni) [inline safe/gui/tools/wizard/step_kw13_band_selector.py]: __[flake8]__
* [F401] 'safe.definitions.utilities.get_layer_modes' imported but unused
* [F401] 'safe.definitions.utilities.definition' imported but unused

File: safe/gui/tools/wizard/step_kw13_band_selector.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+from safe.definitions.layer_modes import layer_mode_continuous
```
Comment (ismailsunni) [inline safe/gui/tools/wizard/step_kw13_band_selector.py]: __[flake8]__
* [F401] 'safe.definitions.layer_modes.layer_mode_continuous' imported but unused

**Comment by ismailsunni** [comment]
Hi @Gustry for your implementation part, just push to this PR so that we put the same single feature in one PR, like @timlinux suggested

**Comment by Gustry** [comment]
ok, working on it !

**Comment by Gustry** [comment]
Not a lot to update, this should be fine, let's wait for travis.

File: safe/gis/raster/zonal_statistics.py [hunk 6]
```diff
@@ -59,8 +59,13 @@ def zonal_stats(raster, vector, callback=None):
 
     copy_layer(vector, layer)
 
+    active_band = layer.keywords.get('active_band', 1)
```
Comment (ismailsunni) [inline safe/gis/raster/zonal_statistics.py]: Why does this one is 1 and the one in poligonize.py is 0?

File: safe/gis/raster/polygonize.py [hunk 7]
```diff
@@ -76,7 +76,9 @@ def polygonize(layer, callback=None):
 
     input_band = input_raster.GetRasterBand(1)
```
Comment (ismailsunni) [inline safe/gis/raster/polygonize.py]: Is it correct to set to 1?

**Comment by Gustry** [comment]
I have updated my latest commit. You can check it.

Should we add the keyword for existing 4.0 keywords? For now, I put 1 as default if the keyword is not found.

**Comment by ismailsunni** [comment]
Hi @Gustry I think it's fine to not add the keyword for the existing 4.0 keywords since in the previous version also use band 1 for the data.

Thanks for the commits, I will merge this one.

**Comment by Gustry** [comment]
What did you do to my commit ??

**Comment by ismailsunni** [comment]
Ouch, I forgot to pull from the branch (I forgot that my branch is out dated), then I rebase from develop, then I push force it. :(
I hope you still have the commit @Gustry

**Pull request merged**